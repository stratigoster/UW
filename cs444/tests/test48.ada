-- no errors here
PACKAGE main is
  i: integer;
END main;

PACKAGE body main IS
  x : integer;
begin
  if true then
    x := 5; -- assignments work!
  elsif false then
    null;
  elsif true then
    null;
  else
    i := 6; -- even found `i' too
  end if;
END main;
