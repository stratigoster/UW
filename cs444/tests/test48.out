

#!/usr/bin/env python
import math
def _forwardGen(seq):
	for i in seq:
		yield { 'all':i }
def _reverseGen(seq):
	i = len(seq)
	while i>0:
		i = i - 1
		yield { 'all':seq[i] }
class _dummy_(Exception):pass
class _numeric_error(Exception):pass
class _constraint_error(Exception):pass
class main:
	try:
		i = { 'all':0 }
		x = { 'all':0 }
		def __init__(self):
			try:
				pass
				if 1:
					self.x['all'] = 5 
				elif 0:
					pass
				elif 1:
					pass
				else:
					self.i['all'] = 6 
			except _dummy_:
				pass
	except _dummy_:
		pass
main()
